### DISCUSSION TITLE
How can I properly activate a piece page type?

### DISCUSSION THREAD

**Discussion by fxpr** [apostrophecms/apostrophe#4176] [Q&A] [answered]
I followed https://v3.docs.apostrophecms.org/guide/piece-pages.html but I do not end up with auto generated index and show pages at `/products` and `/products/productslug`. It is showing me a 404.
Am I missing something?
Any help would be greatly appreciated.

Here are the files I created / changed:
```js
// modules/product/index.js
module.exports = {
  extend: '@apostrophecms/piece-type',
  options: {
    label: 'Product',
    pluralLabel: 'Products'
  },
  fields: {
    add: {
      title: {
        type: 'string',
        label: 'Title',
        required: true
      }
    },
    group: {
      basics: {
        label: 'Basics',
        fields: ['title']
      }
    }
  }
};
```
```js
// modules/product-page/index.js
module.exports = {
  extend: '@apostrophecms/piece-page-type',
  options: {
    label: 'Product Page',
    pluralLabel: 'Product Pages',
    pieceModuleName: 'product'
  },
};
```
```html
<!-- modules/product-page/views/index.html -->
{% extends "layout.html" %}

{% block main %}
  <h1>Product List</h1>
  {% for piece in data.pieces %}
    <article>
      <h2>
        <a href="{{ piece._url }}">{{ piece.title }}</a>
      </h2>
    </article>
  {% endfor %}
{% endblock %}
```
```html
<!-- modules/product-page/views/show.html -->
{% extends "layout.html" %}

{% block main %}
  <h1>{{ data.piece.title }}</h1>
{% endblock %}
```
```js
// app.js
require('apostrophe')({
  shortName: 'myproject',
  modules: {
    '@apostrophecms/rich-text-widget': {
      options: {
        className: 'bp-rich-text'
      }
    },
    '@apostrophecms/image-widget': {
      options: {
        className: 'bp-image-widget'
      }
    },
    '@apostrophecms/video-widget': {
      options: {
        className: 'bp-video-widget'
      }
    },

    'product': {},
    'product-page': {},

    asset: {},
    'default-page': {}
  }
});
```

**Comment by boutell** [comment, marked as answer]
Hmm. I don't see here where you added `product-page` to the list of available page types, or an indication that you then logged into your site and added a page at `/products` for which you chose the type "Product Page."

The documentation page you mention references our page type guide. The section you missed is here:

https://v3.docs.apostrophecms.org/guide/pages.html#activating-page-types

Perhaps we'll add a link to that specific section as a refresher for those in your situation.

**Reply by boutell** [reply]
(Let us know if this solves your problem!)

**Comment by fxpr** [comment]
Yes, this solves it! 

I enjoyed reading the documentation so far but I must have missed that this last step works the same as for `page-type`. In hindsight it's obvious because both are called page types. But since `piece-page-types` enable auto generated `index` and `show` pages for pieces, instead of just custom editable pages, I somehow assumed that I did not have to add a page myself.

Thank you guys for this wonderful project!
